src/webhook/mutation/pod_mutator.go

```diff
@@ -751,14 +752,31 @@ func addMetadataIfMissing(c *corev1.Container, deploymentMetadata *deploymentmet
 			return
 		}
 	}
-
 	c.Env = append(c.Env,
 		corev1.EnvVar{
 			Name:  dynatraceMetadataEnvVarName,
 			Value: deploymentMetadata.AsString(),
 		})
 }
 
+func setInitialConnectRetryIfMissing(c *corev1.Container, dynaKube *dynatracev1beta1.DynaKube) {
+	if dynaKube.FeatureAgentInitialConnectRetry() < 0 {
+		return
+	}
+
+	for _, v := range c.Env {
```

Comment:
To be honest i dont like to force-use generics just because they are `generics`. In that case it gets really hard to read what the code actually does. But we already have a method to check envVars and I used this one for now

Reply:
Reusing already existing code for this purpose - even better :+1: .
Regarding readability, it's just a matter of getting used to it. 
Effectively it's less code to maintain, so maintenance cost is lower and quality is higher (library code is better than my code).